extensions/Push/lib/Push.pm

```diff
@@ -50,15 +38,52 @@ sub start {
         $connector->backlog->reset_backoff();
     }
 
-    while(1) {
-        if ($self->_dbh_check()) {
-            $self->_reload();
-            $self->push();
+    my $pushd_loop = IO::Async::Loop->new;
+    my $main_timer = IO::Async::Timer::Periodic->new(
+        first_interval => 0,
+        interval       => POLL_INTERVAL_SECONDS,
+        reschedule     => 'drift',
+        on_tick        => sub {
+            if ( $self->_dbh_check() ) {
+                $self->_reload();
+                $self->push();
+            }
+        },
+    );
+    if ( Bugzilla->datadog ) {
+        my $dog_timer = IO::Async::Timer::Periodic->new(
+            interval   => 60 * 5,
+            reschedule => 'drift',
+            on_tick    => sub { $self->heartbeat },
+        );
+        $pushd_loop->add($dog_timer);
+        $dog_timer->start;
+    }
+
+    $pushd_loop->add($main_timer);
+    $main_timer->start;
+    $pushd_loop->run;
+}
+
+sub heartbeat {
+    my ($self) = @_;
+    my $dd = Bugzilla->datadog('pushd');
+
+    $dd->gauge('total', $self->_sql_count('SELECT COUNT(*) FROM push'));
```

Comment:
Not sure about the naming of the metric. 'total' will get sent to datadog as 'pushd.total' currently.
@mars-f

Reply:
You want your metric names to be as loose hierarchical namespace with each level separated by dots.  The name `total` is a bit vague for people unfamiliar with how the metric is collected.  `pushd.queue_size` would be clearer.

If there are other pushd components besides the queue that you want to measure, or if you want to measure the queue in multiple ways, then you could make the `queue` level explicit with `pushd.queue.size`. (Personally, I prefer dots instead of underscores - I like how they read.)

If your code already prefixes the `bugzilla` namespace for you then the final metric name will be `bugzilla.pushd.queue.size`.

In prod we already have metrics like `bugzilla.jobqueue.total_errors`.